File: js/nova/datastruct/list/IntArray.js
var IntArray = function () {
	
	
};

IntArray.prototype.IntArray = function () {
	this.this();
};

IntArray.prototype.IntArray = function (count) {
	this.this(count);
};

IntArray.prototype.IntArray = function (data, count) {
	this.this(data, count);
};

IntArray.prototype.destroy = function () {
};

IntArray.prototype.this = function () {
	this.this();
};

IntArray.prototype.this = function (count) {
	this.this(count);
};

IntArray.prototype.this = function (data, count) {
	this.this(data, count);
};

IntArray.prototype.map = function (mapFunc) {
	var array;
	var i;
	var nova_local_0;
	var element;
	array = new Array();
	i = 0;
	nova_local_0 = (this).iterator();
	while (nova_local_0.hasNext()) {
		element = nova_local_0.next();
		array.add(mapFunc(element, i++, this));
	}
	return array;
};

IntArray.prototype.forEach = function (func) {
	var i;
	i = 0;
	for (; i < this.count; i++) {
		func(this.get(i), i, this);
	}
};

IntArray.prototype.any = function (anyFunc) {
	var nova_local_0;
	var element;
	nova_local_0 = (this).iterator();
	while (nova_local_0.hasNext()) {
		element = nova_local_0.next();
		if (anyFunc(element)) {
			return true;
		}
	}
	return false;
};

IntArray.prototype.all = function (allFunc) {
	var nova_local_0;
	var element;
	nova_local_0 = (this).iterator();
	while (nova_local_0.hasNext()) {
		element = nova_local_0.next();
		if (!allFunc(element)) {
			return false;
		}
	}
	return true;
};

IntArray.prototype.filter = function (filterFunc) {
	var filtered;
	var i;
	var nova_local_0;
	var element;
	filtered = new IntArray();
	i = 0;
	nova_local_0 = (this).iterator();
	while (nova_local_0.hasNext()) {
		element = nova_local_0.next();
		if (filterFunc(element, i++, this)) {
			filtered.add(element);
		}
	}
	return filtered;
};

IntArray.prototype.take = function (howMany) {
	var list;
	var i;
	if (howMany > this.count) {
		howMany = this.count;
	}
	list = new IntArray();
	i = 0;
	for (; i < howMany; i++) {
		list.add(this.data);
	}
	return list;
};

IntArray.prototype.skip = function (howMany) {
	var list;
	var i;
	list = new IntArray();
	i = howMany;
	for (; i < this.count; i++) {
		list.add(this.data);
	}
	return list;
};

IntArray.prototype.firstWhere = function (func) {
	var nova_local_0;
	var element;
	nova_local_0 = (this).iterator();
	while (nova_local_0.hasNext()) {
		element = nova_local_0.next();
		if (func(element)) {
			return element;
		}
	}
	return null;
};

IntArray.prototype.reverse = function () {
	var array;
	var i;
	var nova_local_0;
	var element;
	array = new IntArray(this.count);
	array.count = this.count;
	i = 0;
	nova_local_0 = (this).iterator();
	while (nova_local_0.hasNext()) {
		element = nova_local_0.next();
		array.set(this.count - ++i, element);
	}
	return array;
};

IntArray.prototype.join = function (delimiter) {
	var str;
	var passed;
	var nova_local_0;
	var element;
	str = "";
	passed = false;
	nova_local_0 = (this).iterator();
	while (nova_local_0.hasNext()) {
		element = nova_local_0.next();
		if (passed) {
			str = str.concat(delimiter);
		} else {
			passed = true;
		}
		str = str.concat(Int.toString(element));
	}
	return str;
};

IntArray.prototype.accessor_iterator = function () {
	return new IntArrayIterator(this);
};

IntArray.prototype.mutator_iterator = function () {
};

IntArray.prototype.accessor_first = function () {
	if (this.count > 0) {
		return this.data;
	}
	return null;
};

IntArray.prototype.mutator_first = function () {
};

IntArray.prototype.accessor_last = function () {
	if (this.count > 0) {
		return this.data;
	}
	return null;
};

IntArray.prototype.super = function () {
};

File: java/nova/datastruct/list/IntArray.java
package nova.datastruct.list;

import java.util.Optional;
import nova.exception.ExceptionData;
import nova.exception.Exception;
import nova.exception.DivideByZeroException;
import nova.io.NovaConsole;
import nova.primitive.number.Number;
import nova.primitive.number.NovaByte;
import nova.primitive.number.NovaShort;
import nova.primitive.number.NovaInt;
import nova.primitive.number.NovaLong;
import nova.primitive.number.NovaFloat;
import nova.primitive.number.NovaDouble;
import nova.primitive.Null;
import nova.primitive.number.Char;
import nova.primitive.Bool;
import nova.datastruct.list.NovaArray;
import nova.datastruct.list.IntArray;
import nova.datastruct.list.CharArray;
import nova.datastruct.list.DoubleArray;
import nova.datastruct.list.IntRange;
import nova.thread.Thread;
import nova.thread.async.Async;
import nova.gc.GC;
import nova.math.NovaMath;
import nova.NovaObject;
import nova.NovaString;
import nova.System;
import nova.Class;
import nova.datastruct.list.IntArrayIterator;

public class IntArray extends NovaArray
{
	
	
	
	
	
	public IntArray()
	{
		init();
	}
	
	public IntArray(int count)
	{
		init(count);
	}
	
	public IntArray(int[] data, int count)
	{
		init(data, count);
	}
	
	public IntArrayIterator accessor_iterator()
	{
		return new IntArrayIterator(this);
	}
	
	private IntArrayIterator mutator_iterator()
	{
	}
	
	public int accessor_first()
	{
		return count > 0 ? this.get(0) : 0;
	}
	
	private int mutator_first()
	{
	}
	
	public int accessor_last()
	{
		return count > 0 ? this.get(count - 1) : 0;
	}
	
	private int mutator_last()
	{
	}
	
	public void init()
	{
		init();
	}
	
	public void init(int count)
	{
		init(count);
	}
	
	public void init(int[] data, int count)
	{
		init(data, count);
	}
	
	public NovaArray map(NovaUtilities.Function3<Int, Int, IntArray, Out> mapFunc)
	{
		NovaArray array;
		int i;
		IntArrayIterator nova_local_0;
		int element;
		array = new NovaArray();
		i = 0;
		nova_local_0 = (this).accessor_iterator();
		while (nova_local_0.accessor_hasNext())
		{
			element = nova_local_0.accessor_next();
			array.add(mapFunc.call(element, i++, this));
		}
		return array;
	}
	
	public IntArray forEach(NovaUtilities.Consumer3<Int, Int, IntArray> func)
	{
		int i;
		i = (int)0;
		for (; i < (int)count; i++)
		{
			func.call((int)get(i), i, this);
		}
		return this;
	}
	
	public boolean any(NovaUtilities.Function1<Int, Bool> anyFunc)
	{
		IntArrayIterator nova_local_0;
		int element;
		nova_local_0 = (this).accessor_iterator();
		while (nova_local_0.accessor_hasNext())
		{
			element = nova_local_0.accessor_next();
			if (anyFunc.call(element))
			{
				return true;
			}
		}
		return false;
	}
	
	public boolean all(NovaUtilities.Function1<Int, Bool> allFunc)
	{
		IntArrayIterator nova_local_0;
		int element;
		nova_local_0 = (this).accessor_iterator();
		while (nova_local_0.accessor_hasNext())
		{
			element = nova_local_0.accessor_next();
			if (!allFunc.call(element))
			{
				return false;
			}
		}
		return true;
	}
	
	public IntArray filter(NovaUtilities.Function3<Int, Int, IntArray, Bool> filterFunc)
	{
		IntArray filtered;
		int i;
		IntArrayIterator nova_local_0;
		int element;
		filtered = new IntArray();
		i = 0;
		nova_local_0 = (this).accessor_iterator();
		while (nova_local_0.accessor_hasNext())
		{
			element = nova_local_0.accessor_next();
			if (filterFunc.call(element, i++, this))
			{
				filtered.add(element);
			}
		}
		return filtered;
	}
	
	public IntArray take(int howMany)
	{
		IntArray list;
		int i;
		if (howMany > count)
		{
			howMany = count;
		}
		list = new IntArray();
		i = (int)0;
		for (; i < (int)howMany; i++)
		{
			list.add(data[i]);
		}
		return list;
	}
	
	public IntArray skip(int howMany)
	{
		IntArray list;
		int i;
		list = new IntArray();
		i = (int)howMany;
		for (; i < (int)count; i++)
		{
			list.add(data[i]);
		}
		return list;
	}
	
	public int firstWhere(NovaUtilities.Function1<Int, Bool> func)
	{
		IntArrayIterator nova_local_0;
		int element;
		nova_local_0 = (this).accessor_iterator();
		while (nova_local_0.accessor_hasNext())
		{
			element = nova_local_0.accessor_next();
			if (func.call(element))
			{
				return element;
			}
		}
		return null;
	}
	
	public IntArray reverse()
	{
		IntArray array;
		int i;
		IntArrayIterator nova_local_0;
		int element;
		array = new IntArray(count);
		array.count = count;
		i = 0;
		nova_local_0 = (this).accessor_iterator();
		while (nova_local_0.accessor_hasNext())
		{
			element = nova_local_0.accessor_next();
			array.set(count - ++i, element);
		}
		return array;
	}
	
	public NovaString join(NovaString delimiter)
	{
		NovaString str;
		boolean passed;
		IntArrayIterator nova_local_0;
		int element;
		str = new NovaString("");
		passed = false;
		nova_local_0 = (this).accessor_iterator();
		while (nova_local_0.accessor_hasNext())
		{
			element = nova_local_0.accessor_next();
			if (passed)
			{
				str = str.concat(delimiter);
			}
			else
			{
				passed = true;
			}
			str = str.concat(NovaInt.toString(element));
		}
		return str;
	}
	
	public int get(int index)
	{
		return ((int[])data)[index];
	}
	
	public int set(int index, int value)
	{
		((int[])data)[index] = value;
		return value;
	}
	
}
